[Flaky Test] go_test: //staging/src/k8s.io/legacy-cloud-providers/azure/go_default_test:run_1_of_2

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

**Which jobs are flaking**:
bazel-test-master
**Which test(s) are flaking**:
go_test: //staging/src/k8s.io/legacy-cloud-providers/azure/go_default_test:run_1_of_2 

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-blocking#bazel-test-master
**Reason for failure**:
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //staging/src/k8s.io/legacy-cloud-providers/azure:go_default_test
-----------------------------------------------------------------------------
E1015 07:27:27.235219  205663 azure_backoff.go:80] GetVirtualMachineWithRetry(vm): backoff failure, will retry, err=Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.237447  205663 azure_backoff.go:80] GetVirtualMachineWithRetry(vm): backoff failure, will retry, err=Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.237528  205663 azure_backoff.go:122] GetPrivateIPsByNodeName(vm): backoff failure, will retry,err=Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.238934  205663 azure_backoff.go:142] GetIPForMachineWithRetry(vm): backoff failure, will retry,err=instance not found
E1015 07:27:27.251628  205663 azure_backoff.go:216] LoadBalancerClient.List(rg) failure with err=&{false 500 0001-01-01 00:00:00 +0000 UTC <nil>}
E1015 07:27:27.252238  205663 azure_backoff.go:231] PublicIPAddressesClient.List(rg) failure with err=&{false 500 0001-01-01 00:00:00 +0000 UTC <nil>}
E1015 07:27:27.252777  205663 azure_backoff.go:247] PublicIPAddressesClient.CreateOrUpdate(rg, nic) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.253290  205663 azure_backoff.go:263] InterfacesClient.CreateOrUpdate(nic) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.255807  205663 azure_backoff.go:278] PublicIPAddressesClient.Delete(pip) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.256417  205663 azure_backoff.go:304] LoadBalancerClient.Delete(lb) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.257035  205663 azure_backoff.go:331] RouteTablesClient.CreateOrUpdate(rt) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 412, RawError: <nil>
E1015 07:27:27.257470  205663 azure_backoff.go:331] RouteTablesClient.CreateOrUpdate(rt) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: canceledandsupersededduetoanotheroperation
E1015 07:27:27.259234  205663 azure_backoff.go:370] RoutesClient.Delete(rt, rt) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
E1015 07:27:27.259843  205663 azure_backoff.go:384] CreateOrUpdateVMSS: error getting vmss(vmss): &{false 500 0001-01-01 00:00:00 +0000 UTC <nil>}
E1015 07:27:27.260095  205663 azure_backoff.go:384] CreateOrUpdateVMSS: error getting vmss(vmss): &{false 429 0001-01-01 00:00:00 +0000 UTC <nil>}
E1015 07:27:27.260307  205663 azure_backoff.go:384] CreateOrUpdateVMSS: error getting vmss(vmss): &{false 0 0001-01-01 00:00:00 +0000 UTC azure cloud provider rate limited(write) for operation CreateOrUpdate}
E1015 07:27:27.261298  205663 azure_blobDiskController.go:77] azureDisk - getAllStorageAccounts error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
I1015 07:27:27.308920  205663 azure_blobDiskController.go:469] azureDisk - identified account ds-0 as part of shared PVC accounts
W1015 07:27:29.013501  205663 azure_blobDiskController.go:130] azureDisk - failed to delete blob https://foo.blob./vhds/bar.vhd err: storage: service returned error: StatusCode=403, ErrorCode=AccountIsDisabled, ErrorMessage=The specified account is disabled.
RequestId:6b9f7799-601e-0132-3ac4-a24b44000000
Time:2020-10-15T07:27:28.9789324Z, RequestInitiated=Thu, 15 Oct 2020 07:27:28 GMT, RequestId=6b9f7799-601e-0132-3ac4-a24b44000000, API Version=, QueryParameterName=, QueryParameterValue=
I1015 07:27:29.199025  205663 azure_blobDiskController.go:463] azureDisk - accountListResult Name or Sku is nil
I1015 07:27:29.199116  205663 azure_blobDiskController.go:469] azureDisk - identified account ds-0 as part of shared PVC accounts
I1015 07:27:31.638995  205663 azure_blobDiskController.go:596] azureDisk - shared storageAccounts utilization(0.9835) > grow-at-avg-utilization (0.5). But k8s maxed on SAs for PVC(100). k8s will now exceed grow-at-avg-utilization without adding accounts
--- FAIL: TestDeleteBlobDisk (183.09s)
    azure_blobDiskController_test.go:395: 
        	Error Trace:	azure_blobDiskController_test.go:395
        	Error:      	Should be true
        	Test:       	TestDeleteBlobDisk
E1015 07:30:34.975368  205663 azure_config.go:62] Failed to initialize Azure cloud provider: cloudConfigType UnknownConfigType is not supported, supported values are [file merge secret]
W1015 07:30:34.987169  205663 azure_config.go:52] Failed to get cloud-config from secret: failed to get secret azure-cloud-provider: secrets "azure-cloud-provider" not found, skip initializing from secret
W1015 07:30:34.987572  205663 azure_config.go:52] Failed to get cloud-config from secret: cloud-config is not set in the secret (azure-cloud-provider), skip initializing from secret
W1015 07:30:34.988389  205663 azure_controller_common.go:186] failed to get azure instance id (instance not found) for node vm1
W1015 07:30:34.989287  205663 azure_controller_common.go:195] no LUN available for instance "vm1" (all luns are used)
W1015 07:30:34.989684  205663 azure_controller_common.go:167] size of disk(/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) is 4096GB which is bigger than limit(4096GB), set cacheMode as None
W1015 07:30:34.992750  205663 azure_controller_common.go:186] failed to get azure instance id (not a vmss instance) for node vmss1
W1015 07:30:34.993252  205663 azure_controller_common.go:211] azureDisk - failed to get azure instance id("vm1"), DetachDisk(/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) will assume disk is already detached
E1015 07:30:34.993989  205663 azure_controller_standard.go:156] detach azure disk: disk  not found, diskURI: /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name
E1015 07:30:34.994066  205663 azure_controller_standard.go:175] azureDisk - detach disk(, /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) on rg(rg) vm(vm1) failed, err: &{true 400 0001-01-01 00:00:00 +0000 UTC Retriable: true}
W1015 07:30:34.994256  205663 azure_controller_common.go:241] azureDisk - update backing off: detach disk(, /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name), err: Retriable: true, RetryAfter: 0s, HTTPStatusCode: 400, RawError: Retriable: true
E1015 07:30:34.994367  205663 azure_controller_standard.go:156] detach azure disk: disk  not found, diskURI: /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name
E1015 07:30:34.994432  205663 azure_controller_standard.go:175] azureDisk - detach disk(, /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) on rg(rg) vm(vm1) failed, err: &{true 400 0001-01-01 00:00:00 +0000 UTC Retriable: true}
E1015 07:30:34.994562  205663 azure_controller_common.go:262] azureDisk - detach disk(, /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) failed, err: Retriable: true, RetryAfter: 0s, HTTPStatusCode: 400, RawError: Retriable: true
E1015 07:30:34.995216  205663 azure_controller_standard.go:156] detach azure disk: disk disk2 not found, diskURI: /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name
W1015 07:30:35.003014  205663 azure_controller_common.go:343] azureDisk - Cannot find node "vm2", DisksAreAttached will assume disks [disk1] are not attached to it.
E1015 07:30:35.006525  205663 azure_controller_common.go:389] disk(/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/DiskName2) does not exist, removed from data disk list
E1015 07:30:35.006650  205663 azure_controller_common.go:389] disk(/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/DiskName3) does not exist, removed from data disk list
E1015 07:30:35.006762  205663 azure_controller_common.go:389] disk(/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/DiskName4) does not exist, removed from data disk list
E1015 07:30:35.007095  205663 azure_controller_common.go:384] checkDiskExists(/subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/specialdisk) failed with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 400, RawError: instance not found
E1015 07:30:35.008481  205663 azure_controller_standard.go:102] azureDisk - attach disk(, uri) on rg(rg) vm(vm1) failed, err: &{false 404 0001-01-01 00:00:00 +0000 UTC instance not found}
E1015 07:30:35.008584  205663 azure_controller_standard.go:104] azureDisk - begin to filterNonExistingDisks(, uri) on rg(rg) vm(vm1)
E1015 07:30:35.008625  205663 azure_controller_common.go:384] checkDiskExists(uri) failed with error: invalid disk URI: uri
W1015 07:30:35.009147  205663 azure_controller_standard.go:124] azureDisk - cannot find node vm2, skip detaching disk(, )
E1015 07:30:35.009418  205663 azure_controller_standard.go:156] detach azure disk: disk disk2 not found, diskURI: 
E1015 07:30:35.010003  205663 azure_controller_standard.go:156] detach azure disk: disk  not found, diskURI: 
E1015 07:30:35.010076  205663 azure_controller_standard.go:175] azureDisk - detach disk(, ) on rg(rg) vm(vm1) failed, err: &{false 404 0001-01-01 00:00:00 +0000 UTC instance not found}
E1015 07:30:35.010155  205663 azure_controller_standard.go:177] azureDisk - begin to filterNonExistingDisks(, ) on rg(rg) vm(vm1)
E1015 07:30:37.013843  205663 azure_controller_vmss.go:104] azureDisk - attach disk(disk-name, /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) on rg(rg) vm(vmss00-vm-000000) failed, err: &{false 404 0001-01-01 00:00:00 +0000 UTC instance not found}
E1015 07:30:37.013961  205663 azure_controller_vmss.go:106] azureDisk - begin to filterNonExistingDisks(disk-name, /subscriptions/subscription/resourceGroups/rg/providers/Microsoft.Compute/disks/disk-name) on rg(rg) vm(vmss00-vm-000000)
E1015 07:30:37.015523  205663 azure_controller_vmss.go:178] azureDisk - detach disk(disk-name, disk-name) on rg(rg) vm(vmss00-vm-000000) failed, err: &{false 404 0001-01-01 00:00:00 +0000 UTC instance not found}
E1015 07:30:37.015610  205663 azure_controller_vmss.go:180] azureDisk - begin to filterNonExistingDisks(disk-name, disk-name) on rg(rg) vm(vmss00-vm-000000)
E1015 07:30:37.016132  205663 azure_controller_vmss.go:158] detach azure disk: disk disk-name-err not found, diskURI: disk-name
E1015 07:30:37.077545  205663 azure_backoff.go:142] GetIPForMachineWithRetry(vm2): backoff failure, will retry,err=instance not found
E1015 07:30:37.101887  205663 azure_vmss.go:699] VirtualMachineScaleSetVMsClient.List failed: &{false 404 0001-01-01 00:00:00 +0000 UTC <nil>}
E1015 07:30:37.139881  205663 azure_loadbalancer.go:175] reconcileLoadBalancer(default/service4) failed: findMatchedPIPByLoadBalancerIP: cannot find public IP with IP address 1.2.3.4 in resource group random-rg
E1015 07:30:37.144071  205663 azure_loadbalancer.go:204] reconcilePublicIP(default/service5) failed: &errors.errorString{s:"Retriable: false, RetryAfter: 0s, HTTPStatusCode: 404, RawError: instance not found"}
E1015 07:30:37.145198  205663 azure_backoff.go:216] LoadBalancerClient.List(nil) failure with err=&{false 404 0001-01-01 00:00:00 +0000 UTC instance not found}
E1015 07:30:37.152731  205663 azure_backoff.go:216] LoadBalancerClient.List(nil) failure with err=&{false 404 0001-01-01 00:00:00 +0000 UTC instance not found}
E1015 07:30:37.152933  205663 azure_loadbalancer.go:995] reconcileLoadBalancer: failed to get load balancer for service "default/service5", error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 404, RawError: instance not found
I1015 07:30:37.161156  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service1) isInternal(false) - vmSetNames [as as-1 as-2]
W1015 07:30:37.166426  205663 azure_loadbalancer.go:1459] annotation service.beta.kubernetes.io/azure-load-balancer-disable-tcp-reset has been removed as of Kubernetes 1.20. TCP Resets are always enabled on Standard SKU load balancers.
W1015 07:30:37.179718  205663 azure_loadbalancer.go:1459] annotation service.beta.kubernetes.io/azure-load-balancer-disable-tcp-reset has been removed as of Kubernetes 1.20. TCP Resets are always enabled on Standard SKU load balancers.
I1015 07:30:37.189499  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service1) isInternal(false) - vmSetNames [as]
I1015 07:30:37.189860  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1) - adding
I1015 07:30:37.193265  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service1) isInternal(false) - vmSetNames [as]
I1015 07:30:37.193716  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1) - adding
I1015 07:30:37.197078  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1) - adding
W1015 07:30:37.216606  205663 azure_loadbalancer.go:2437] cannot find the service ns2/svc2 in the corresponding PIP
W1015 07:30:37.216851  205663 azure_loadbalancer.go:2437] cannot find the service ns2/svc2 in the corresponding PIP
W1015 07:30:37.217113  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB lb cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:37.217184  205663 azure_loadbalancer.go:916] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB lb cannot be deleted because it is being referenced by the outbound rule aservice1-rule
W1015 07:30:37.217262  205663 azure_loadbalancer.go:932] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB lb cannot be deleted because it is being referenced by the inbound NAT rule aservice2-rule
W1015 07:30:37.217320  205663 azure_loadbalancer.go:947] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB lb cannot be deleted because it is being referenced by the inbound NAT pool aservice2-rule
E1015 07:30:37.221790  205663 azure_managedDiskController.go:365] Failed to get information for AzureDisk "fakeGetDiskFailed": &{false 0 0001-01-01 00:00:00 +0000 UTC Get Disk failed}
E1015 07:30:37.222115  205663 azure_managedDiskController.go:349] Failed to get resource group for AzureDisk "invalidDiskURI": invalid disk URI: invalidDiskURI
E1015 07:30:37.436569  205663 azure_backoff.go:331] RouteTablesClient.CreateOrUpdate(rt2) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: CreateOrUpdate error
E1015 07:30:37.436718  205663 azure_routes.go:186] CreateOrUpdateRouteTable() failed with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: CreateOrUpdate error
E1015 07:30:37.436798  205663 azure_routes.go:356] CreateRoute failed for node "node" with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: CreateOrUpdate error
E1015 07:30:37.636468  205663 azure_backoff.go:331] RouteTablesClient.CreateOrUpdate(rt4) failed: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: CreateOrUpdate error
E1015 07:30:37.636587  205663 azure_routes.go:136] createRouteTable() failed with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: CreateOrUpdate error
E1015 07:30:37.636659  205663 azure_routes.go:356] CreateRoute failed for node "node" with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: CreateOrUpdate error
E1015 07:30:37.736736  205663 azure_routes.go:142] getRouteTable() failed with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: Get error
E1015 07:30:37.736850  205663 azure_routes.go:356] CreateRoute failed for node "node" with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: Get error
E1015 07:30:37.836249  205663 azure_routes.go:128] getRouteTable() failed with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: Get error
E1015 07:30:37.836379  205663 azure_routes.go:356] CreateRoute failed for node "node" with error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: Get error
E1015 07:30:37.836912  205663 azure_backoff.go:142] GetIPForMachineWithRetry(node): backoff failure, will retry,err=getIP error
E1015 07:30:37.944394  205663 azure_standard.go:466] convertResourceGroupNameToLower failed with error: "/subscriptions/subscription/resourceGroups/rg/Microsoft.Compute/virtualMachines/vm4" isn't in Azure resource ID format ".*/subscriptions/(?:.*)/resourceGroups/(.+)/providers/(?:.*)"
E1015 07:30:37.946217  205663 azure_standard.go:632] as.getNodeAvailabilitySet - Node(vm2) has no availability sets
E1015 07:30:37.946297  205663 azure_standard.go:667] as.GetVMSetNames - getAgentPoolAvailabiliySets failed err=(Node (vm2) - has no availability sets)
E1015 07:30:37.946421  205663 azure_standard.go:691] as.GetVMSetNames - Availability set (vm2) in service annotation not found
E1015 07:30:37.948358  205663 azure_standard.go:782] error: az.EnsureHostInPool(vm2), az.VMSet.GetPrimaryInterface.Get(vm2, availabilityset-1), err=resource name was missing from identifier
W1015 07:30:37.949020  205663 azure_standard.go:787] EnsureHostInPool skips node vm3 because its primary nic nic3 is in Failed state
E1015 07:30:37.965608  205663 azure_standard.go:356] serviceOwnsFrontendIP: unexpected error when finding match public IP of the service  with loadBalancerLP 1.2.3.4
I1015 07:30:38.088473  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(true) - vmSetNames [as]
E1015 07:30:38.088613  205663 azure_loadbalancer.go:402] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(true) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
E1015 07:30:38.088705  205663 azure_loadbalancer.go:995] reconcileLoadBalancer: failed to get load balancer for service "default/service-2", error: selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(true) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
E1015 07:30:38.088759  205663 azure_loadbalancer.go:175] reconcileLoadBalancer(default/service-2) failed: selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(true) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
W1015 07:30:38.115794  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB testCluster-internal cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.116438  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB as-3-internal cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.117408  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB as-2-internal cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.118304  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB as-1-internal cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.119658  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB testCluster-internal cannot be deleted because it is being referenced by load balancing rules of other services
I1015 07:30:38.258003  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(false) - vmSetNames [as]
E1015 07:30:38.258087  205663 azure_loadbalancer.go:402] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(false) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
E1015 07:30:38.258129  205663 azure_loadbalancer.go:995] reconcileLoadBalancer: failed to get load balancer for service "default/service-2", error: selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(false) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
E1015 07:30:38.258173  205663 azure_loadbalancer.go:175] reconcileLoadBalancer(default/service-2) failed: selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(false) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
W1015 07:30:38.323361  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB testCluster cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.325581  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB as-3 cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.327495  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB as-2 cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.329432  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB as-1 cannot be deleted because it is being referenced by load balancing rules of other services
W1015 07:30:38.330684  205663 azure_loadbalancer.go:901] isFrontendIPConfigIsUnsafeToDelete: frontend IP configuration with ID fip on LB testCluster cannot be deleted because it is being referenced by load balancing rules of other services
I1015 07:30:38.345909  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1-subnet) - adding
I1015 07:30:38.369475  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service-sa-none)(true): lb frontendconfig(aservicesanone) - adding
I1015 07:30:38.371682  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service-sa-clientip)(true): lb frontendconfig(aservicesaclientip) - adding
I1015 07:30:38.379185  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service1) isInternal(false) - vmSetNames [as]
I1015 07:30:38.379585  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1) - adding
I1015 07:30:38.380633  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service2) isInternal(false) - vmSetNames [as]
I1015 07:30:38.381243  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service2)(true): lb frontendconfig(aservice2) - adding
I1015 07:30:38.384350  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service1) isInternal(false) - vmSetNames [as]
I1015 07:30:38.384768  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1) - adding
I1015 07:30:38.390663  205663 azure_loadbalancer.go:361] selectLoadBalancer: cluster(testCluster) service(default/service1) isInternal(false) - vmSetNames [as]
I1015 07:30:38.391076  205663 azure_loadbalancer.go:1170] reconcileLoadBalancer for service (default/service1)(true): lb frontendconfig(aservice1) - adding
I1015 07:30:39.175046  205663 azure.go:720] Setting up informers for Azure cloud provider
W1015 07:30:39.176089  205663 azure.go:343] cloud-config is not provided, Azure cloud provider would be initialized from secret
E1015 07:30:41.195717  205663 azure_vmss.go:818] error: ss.GetPrimaryInterface(agente6c2000005), ss.getVmssVM(agente6c2000005), err=instance not found
E1015 07:30:41.195834  205663 azure_vmss.go:456] error: ss.GetIPByNodeName(agente6c2000005), GetPrimaryInterface("agente6c2000005"), err=instance not found
E1015 07:30:41.203718  205663 azure_vmss_cache.go:73] VirtualMachineScaleSetsClient.List failed: &{false 0 0001-01-01 00:00:00 +0000 UTC error during vmss list}
E1015 07:30:41.210033  205663 azure_backoff.go:99] VirtualMachinesClient.List(rg) failure with err=&{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.210250  205663 azure_vmss.go:382] Failed to check isNodeManagedByAvailabilitySet: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: error
E1015 07:30:41.214097  205663 azure_backoff.go:99] VirtualMachinesClient.List(rg) failure with err=&{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.214283  205663 azure_vmss.go:803] Failed to check isNodeManagedByAvailabilitySet: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: error
E1015 07:30:41.215020  205663 azure_vmss_cache.go:73] VirtualMachineScaleSetsClient.List failed: &{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.215186  205663 azure_vmss.go:818] error: ss.GetPrimaryInterface(vmss-vm-000000), ss.getVmssVM(vmss-vm-000000), err=Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: error
E1015 07:30:41.216371  205663 azure_vmss.go:824] error: ss.GetPrimaryInterface(vmss-vm-000000), ss.getPrimaryInterfaceID(), err=failed to find a primary nic for the vm. vmname="vmss_0"
E1015 07:30:41.217624  205663 azure_vmss.go:830] error: ss.GetPrimaryInterface(vmss-vm-000000), getLastSegment(invalid/id/), err=resource name was missing from identifier
E1015 07:30:41.219850  205663 azure_vmss.go:844] error: ss.GetPrimaryInterface(vmss-vm-000000), ss.GetVirtualMachineScaleSetNetworkInterface.Get(rg, vmss, vmss-vm-000000), err=&{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.230241  205663 azure_backoff.go:99] VirtualMachinesClient.List(rg) failure with err=&{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.230537  205663 azure_vmss.go:803] Failed to check isNodeManagedByAvailabilitySet: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: error
E1015 07:30:41.230634  205663 azure_vmss.go:519] error: ss.GetIPByNodeName(vmss-vm-000000), GetPrimaryInterface("vmss-vm-000000"), err=Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: error
E1015 07:30:41.234731  205663 azure_vmss.go:609] VirtualMachineScaleSetsClient.List failed: &{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.235564  205663 azure_vmss.go:699] VirtualMachineScaleSetVMsClient.List failed: &{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.262743  205663 azure_vmss.go:779] ss.GetVMSetNames - scale set (vmss-1) in service annotation not found
E1015 07:30:41.289513  205663 azure_vmss.go:1236] EnsureHostInPool(/): backendPoolID(/subscriptions/sub/resourceGroups/rg/providers/Microsoft.Network/loadBalancers/lb/backendAddressPools/backendpool-1) - failed to ensure host in pool: "instance not found"
E1015 07:30:41.303820  205663 azure_backoff.go:395] CreateOrUpdateVMSS: error CreateOrUpdate vmss(vmss): &{false 0 0001-01-01 00:00:00 +0000 UTC error}
E1015 07:30:41.304025  205663 azure_vmss.go:1483] ensureBackendPoolDeletedFromVMSS CreateOrUpdateVMSS(vmss) with new backendPoolID /subscriptions/sub/resourceGroups/rg/providers/Microsoft.Network/loadBalancers/lb/backendAddressPools/backendpool-0, err: <nil>
E1015 07:30:41.310992  205663 azure_vmss.go:1579] EnsureBackendPoolDeleted UpdateVMs for VMSS(rg, vmss) failed with error Retriable: false, RetryAfter: 0s, HTTPStatusCode: 0, RawError: error
E1015 07:30:41.314607  205663 azure_vmss.go:1538] Failed to getNodeNameByIPConfigurationID(/subscriptions/sub/resourceGroups/rg1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss-0/virtualMachines/0/networkInterfaces/nic): instance not found
FAIL
```


**Anything else we need to know**:
- links to go.k8s.io/triage appreciated
https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=%2F%2Fstaging%2Fsrc%2Fk8s.io%2Flegacy-cloud-providers%2Fazure

- links to specific failures in spyglass appreciated
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1316634244088336384
<!-- Please see the deflaking doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md) for more guidance! -->

/sig cloud-provider
@kubernetes/ci-signal

Interestingly the expected test string `storage: service returned error: StatusCode=403, ErrorCode=AccountIsDisabled, ErrorMessage=The specified account is disabled.` was logged:
```
E1015 07:27:27.261298  205663 azure_blobDiskController.go:77] azureDisk - getAllStorageAccounts error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>
I1015 07:27:27.308920  205663 azure_blobDiskController.go:469] azureDisk - identified account ds-0 as part of shared PVC accounts
W1015 07:27:29.013501  205663 azure_blobDiskController.go:130] azureDisk - failed to delete blob https://foo.blob./vhds/bar.vhd err: storage: service returned error: StatusCode=403, ErrorCode=AccountIsDisabled, ErrorMessage=The specified account is disabled.
RequestId:6b9f7799-601e-0132-3ac4-a24b44000000
Time:2020-10-15T07:27:28.9789324Z, RequestInitiated=Thu, 15 Oct 2020 07:27:28 GMT, RequestId=6b9f7799-601e-0132-3ac4-a24b44000000, API Version=, QueryParameterName=, QueryParameterValue=
I1015 07:27:29.199025  205663 azure_blobDiskController.go:463] azureDisk - accountListResult Name or Sku is nil
I1015 07:27:29.199116  205663 azure_blobDiskController.go:469] azureDisk - identified account ds-0 as part of shared PVC accounts
I1015 07:27:31.638995  205663 azure_blobDiskController.go:596] azureDisk - shared storageAccounts utilization(0.9835) > grow-at-avg-utilization (0.5). But k8s maxed on SAs for PVC(100). k8s will now exceed grow-at-avg-utilization without adding accounts
--- FAIL: TestDeleteBlobDisk (183.09s)
    azure_blobDiskController_test.go:395: 
        	Error Trace:	azure_blobDiskController_test.go:395
        	Error:      	Should be true
        	Test:       	TestDeleteBlobDisk
```

Test: 

https://github.com/kubernetes/kubernetes/blob/53d63fb748168f18396eee29cabd62815d30da95/staging/src/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go#L393-L395

--- a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/azure/azure_blobDiskController_test.go
@@ -358,39 +358,3 @@ func TestCreateBlobDisk(t *testing.T) {
 	assert.True(t, strings.Contains(err.Error(), expectedErr))
 	assert.Empty(t, diskURI)
 }
-
-func TestDeleteBlobDisk(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-	b := GetTestBlobDiskController(t)
-	b.common.cloud.BlobDiskController = &b
-
-	mockSAClient := mockstorageaccountclient.NewMockInterface(ctrl)
-	b.common.cloud.StorageAccountClient = mockSAClient
-
-	fakeDiskURL := "fake"
-	diskURL := "https://foo.blob./vhds/bar.vhd"
-
-	err := b.DeleteBlobDisk(fakeDiskURL)
-	expectedErr := fmt.Errorf("failed to parse vhd URI invalid vhd URI for regex https://(.*).blob./vhds/(.*): fake")
-	assert.Equal(t, expectedErr, err)
-
-	mockSAClient.EXPECT().ListKeys(gomock.Any(), b.common.resourceGroup, "foo").Return(storage.AccountListKeysResult{}, &retryError500)
-	err = b.DeleteBlobDisk(diskURL)
-	expectedErr = fmt.Errorf("no key for storage account foo, err Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>")
-	assert.Equal(t, expectedErr, err)
-
-	mockSAClient.EXPECT().ListKeys(gomock.Any(), b.common.resourceGroup, "foo").Return(storage.AccountListKeysResult{
-		Keys: &[]storage.AccountKey{
-			{
-				KeyName: to.StringPtr("key1"),
-				Value:   to.StringPtr("key1"),
-			},
-		},
-	}, nil)
-	b.accounts["foo"] = &storageAccountState{}
-	err = b.DeleteBlobDisk(diskURL)
-	expectedErrStr := "storage: service returned error: StatusCode=403, ErrorCode=AccountIsDisabled, ErrorMessage=The specified account is disabled."
-	assert.Error(t, err)
-	assert.True(t, strings.Contains(err.Error(), expectedErrStr))
-}

